### PR TITLE
Greatly speed up RecentDiscussionThread

### PR DESCRIPTION
**Pull request by jimrandomh**
This changes RecentDiscussionThread so that, rather than having a component which uses withList to load 20 posts, and each of those posts has a component which loads some comments, there is a resolver on the Posts collection which gets those comments in a single pass. If I didn't mess this up, this should have indistinguishable end-user behavior, except for one difference: It's much faster.

This removes one server round-trip in the process of loading recent discussion, and saves somewhere between half a second and a second of client-side Javascript execution time (debug mode). The savings come form a combination of fewer rerender passes, and fewer graphql queries (the per-query overhead, it turns out, is quite significant.)

### CODE REVIEW

**Comment by Discordius** [comment]
Lint

File: packages/vulcan-core/lib/modules/containers/withMulti.js [hunk 1]
```diff
@@ -63,11 +63,17 @@ export default function withMulti(options) {
   //pollInterval = typeof window === 'undefined' ? null : pollInterval;
 
   const { collectionName, collection } = extractCollectionInfo(options);
-  const { fragmentName, fragment, extraVariablesString } = extractFragmentInfo(options, collectionName);
+  const { fragmentName, fragment } = extractFragmentInfo(options, collectionName);
 
   const typeName = collection.options.typeName;
   const resolverName = collection.options.multiResolverName;
 
+  // LESSWRONG MODIFICATION: Allow the passing of extraVariables so that you can have field-specific queries
```
Comment (Discordius) [inline packages/vulcan-core/lib/modules/containers/withMulti.js]: Hmm, you seem to make this modification only in withMulti, which means it likely doesn't work in `withSingle`, `withDelete`, `withCreate`, etc.?

File: packages/lesswrong/components/common/HomeEA.jsx [hunk 2]
```diff
@@ -20,7 +21,12 @@ class HomeEA extends PureComponent {
         }
 
         <HomeLatestPosts />
-        <RecentDiscussionThreadsList terms={{view: 'recentDiscussionThreadsList', limit:20}}/>
+        <RecentDiscussionThreadsList
+          terms={{view: 'recentDiscussionThreadsList', limit:20}}
+          commentsLimit={recentDiscussionCommentsPerPost}
+          maxAgeHours={18}
+          af={false}
```
Comment (Raemon) [inline packages/lesswrong/components/common/HomeEA.jsx]: I'd have "af" be false by default, no reason every forum should need to specifiy this

File: packages/lesswrong/components/ea-forum/EAHome.jsx [hunk 3]
```diff
@@ -2,16 +2,22 @@ import { Components, registerComponent } from 'meteor/vulcan:core';
...
       <HomeLatestPosts />
 
       <SingleColumnSection>
         <SectionTitle title="Recent Discussion" />
-        <RecentDiscussionThreadsList terms={{view: 'recentDiscussionThreadsList', limit:6}}/>
+        <RecentDiscussionThreadsList
+          terms={{view: 'recentDiscussionThreadsList', limit:6}}
+          commentsLimit={recentDiscussionCommentsPerPost}
+          maxAgeHours={18}
+          af={false}
```
Comment (Raemon) [inline packages/lesswrong/components/ea-forum/EAHome.jsx]: same here as HomeEA

File: packages/vulcan-core/lib/modules/containers/withMulti.js (hunk 1 above)
Comment (jimrandomh) [reply packages/vulcan-core/lib/modules/containers/withMulti.js]: It already existed in `withSingle`; this is more-or-less a copy-paste of that

File: packages/lesswrong/components/common/HomeEA.jsx (hunk 2 above)
Comment (jimrandomh) [reply packages/lesswrong/components/common/HomeEA.jsx]: `af` isn't going to `RecentDiscussionThreadsList`, it's going to its `withMulti` HoC, which doesn't have a mechanism for specifying default values.

File: packages/lesswrong/components/ea-forum/EAHome.jsx (hunk 3 above)
Comment (jimrandomh) [reply packages/lesswrong/components/ea-forum/EAHome.jsx]: See above

File: packages/vulcan-core/lib/modules/containers/withMulti.js (hunk 1 above)
Comment (Discordius) [reply packages/vulcan-core/lib/modules/containers/withMulti.js]: Ah, cool. Then that seems good.

File: packages/lesswrong/components/comments/RecentDiscussionThread.jsx [hunk 4]
```diff
@@ -177,7 +170,6 @@ class RecentDiscussionThread extends PureComponent {
                   children={comment.children}
                   key={comment.item._id}
                   updateComment={updateComment}
-                  refetch={refetch}
```
Comment (Raemon) [inline packages/lesswrong/components/comments/RecentDiscussionThread.jsx]: I'm not sure if refetch still naively works here, but note that the current version of this PR breaks the functionality that it was doing (i.e. if you reply to a comment in recent discussion, the comment now disappears until you refresh the page, which is what the refetch was supposed do address)

**Pull request merged**